exercises/01-composite-ui/before/Divergent.Sales.API/Controllers/OrdersController.cs

```diff
@@ -1,35 +1,32 @@
 ﻿using System.Collections.Generic;
-using System.Data.Entity;
-using System.Linq;
 using System.Web.Http;
-using Divergent.Sales.Data.Context;
+using Divergent.Sales.Data.Repositories;
+using System.Linq;
 
 namespace Divergent.Sales.API.Controllers
 {
     [RoutePrefix("api/orders")]
     public class OrdersController : ApiController
     {
-        private readonly ISalesContext _context;
+        private readonly IOrderRepository _repository;
 
-        public OrdersController(ISalesContext context)
+        public OrdersController(IOrderRepository repository)
         {
-            _context = context;
+            _repository = repository;
         }
 
         [HttpGet]
         public IEnumerable<dynamic> Get()
         {
-            var orders = _context.Orders
-                .Include(i => i.Items)
-                .Include(i => i.Items.Select(x => x.Product))
-                .ToArray();
+            var orders = _repository.Orders().Result;
```

Comment:
This method should be simply

```C#
public Task<List<Order>> Get() => _repository.Orders();
```

See `CustomersController` for reference. Note that we should not be mapping the `Order` object into a new anonymous type any longer. It should be a simple . This may result in some changes to the view model.

Reply:
As discussed, I missed `ItemsCount = o.Items.Count` so we need to keep the mapping. The only required change is to change the method to be `async`.